app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue

```diff
@@ -0,0 +1,202 @@
+              {{ $t('CAMPAIGN.ADD.FORM.END_POINT.ERROR') }}
+            </span>
+          </label>
+        </div>
+        <div class="medium-12 columns">
+          <label :class="{ error: $v.timeOnPage.$error }">
+            {{ $t('CAMPAIGN.ADD.FORM.TIME_ON_PAGE.LABEL') }}
+            <input
+              v-model.trim="timeOnPage"
+              type="number"
+              :placeholder="$t('CAMPAIGN.ADD.FORM.TIME_ON_PAGE.PLACEHOLDER')"
+              @input="$v.timeOnPage.$touch"
+            />
+            <span v-if="$v.timeOnPage.$error" class="message">
+              {{ $t('CAMPAIGN.ADD.FORM.TIME_ON_PAGE.ERROR') }}
+            </span>
+          </label>
+        </div>
+
+        <div class="medium-12 columns">
+          <label>
+            <input
+              v-model="enabled"
+              type="checkbox"
+              value="enabled"
+              name="enabled"
+            />
+            {{ $t('CAMPAIGN.ADD.FORM.ENABLED') }}
+          </label>
+        </div>
+
+        <div class="modal-footer">
+          <div class="medium-12 columns">
+            <woot-submit-button
+              :disabled="
+                $v.message.$invalid ||
+                $v.title.$invalid ||
+                $v.selectedAgent.$invalid ||
+                $v.endPoint.$invalid ||
+                $v.timeOnPage.$invalid ||
+                uiFlags.isCreating
+              "
+              :loading="uiFlags.isCreating"
+              :button-text="$t('CAMPAIGN.ADD.CREATE_BUTTON_TEXT')"
+            />
+            <button class="button clear" @click.prevent="onClose">
+              {{ $t('CAMPAIGN.ADD.CANCEL_BUTTON_TEXT') }}
+            </button>
+          </div>
+        </div>
+      </form>
+    </div>
+  </modal>
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+import { required, url, minLength } from 'vuelidate/lib/validators';
+
+import WootSubmitButton from '../../../../../components/buttons/FormSubmitButton';
+import Modal from '../../../../../components/Modal';
+import alertMixin from 'shared/mixins/alertMixin';
+
+export default {
+  components: {
+    WootSubmitButton,
+    Modal,
+  },
+  mixins: [alertMixin],
+  props: {
+    onClose: {
+      type: Function,
+      default: () => {},
+    },
+  },
+  data() {
+    return {
+      title: '',
+      message: '',
+      selectedAgent: '',
+      endPoint: '',
+      timeOnPage: 10,
+      show: true,
+      enabled: true,
+    };
+  },
+  validations: {
+    title: {
+      required,
+    },
+    message: {
+      required,
+    },
+    selectedAgent: {
+      required,
+    },
+    endPoint: {
+      required,
+      minLength: minLength(7),
+      url,
+    },
+    timeOnPage: {
+      required,
+    },
+  },
+
+  computed: {
+    ...mapGetters({
+      agentList: 'agents/getAgents',
+      uiFlags: 'campaigns/getUIFlags',
+    }),
+
+    agentsList() {
+      return this.agentList;
+    },
+  },
+  methods: {
+    addCampaign() {
+      this.$store
```

Comment:
We are mostly following async await for this syntax. Can we use the same if it's possible here ?